### PR TITLE
New CrossConnection commandlets and additional parameters to BgpPeering commandlets

### PR DESCRIPTION
**Pull request by nupurkot**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@nupurkot__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (nkothar). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by markcowl** [comment]
@nupurkot tests failed

**Comment by stankovski** [comment]
@azuresdkci retest this please

File: src/ServiceManagement/Services/Commands.Test/Commands.Test.csproj [hunk 1]
```diff
@@ -318,6 +318,7 @@
       <SubType>Designer</SubType>
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </Content>
+    <None Include="app.config" />
```
Comment (stankovski) [inline src/ServiceManagement/Services/Commands.Test/Commands.Test.csproj]: Please remove app.config from csproj file

Comment (nupurkot) [inline src/ServiceManagement/Services/Commands.Test/Commands.Test.csproj]: Fixed

File: setup/azurepowershell.sln [hunk 2]
```diff
@@ -1,7 +1,5 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 2013
-VisualStudioVersion = 12.0.30110.0
-MinimumVisualStudioVersion = 10.0.40219.1
+# Visual Studio 2012
```
Comment (ogail) [inline setup/azurepowershell.sln]: Revert these changes

File: src/ServiceManagement/ExpressRoute/Commands.ExpressRoute/CrossConnections/GetAzureCrossConnection.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+namespace Microsoft.WindowsAzure.Commands.ExpressRoute
+{
+    [Cmdlet(VerbsCommon.Get, "AzureCrossConnection"), OutputType(typeof(AzureCrossConnection), typeof(IEnumerable<AzureCrossConnection>))]
+    public class GetAzureCrossConnectionCommand : ExpressRouteBaseCmdlet
+    {
+        [Parameter(Position = 0, Mandatory = false, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Service Key representing the Dedicated Circuit")]
+        [ValidateGuid]
+        [ValidateNotNullOrEmpty]
+        public string ServiceKey { get; set; }
```
Comment (ogail) [inline src/ServiceManagement/ExpressRoute/Commands.ExpressRoute/CrossConnections/GetAzureCrossConnection.cs]: You may consider making this of type Guid

File: src/ServiceManagement/ExpressRoute/Commands.ExpressRoute/CrossConnections/SetAzureCrossConnection.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        public string ProvisioningError { get; set; }
+
+		[Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            var crossConnection = ExpressRouteClient.SetAzureCrossConnection(ServiceKey,
+                new CrossConnectionUpdateParameters() {Operation = Operation, ProvisioningError = ProvisioningError});
+            if(PassThru.IsPresent)
```
Comment (ogail) [inline src/ServiceManagement/ExpressRoute/Commands.ExpressRoute/CrossConnections/SetAzureCrossConnection.cs]: nit: fix aignment

Comment (nupurkot) [inline src/ServiceManagement/ExpressRoute/Commands.ExpressRoute/CrossConnections/SetAzureCrossConnection.cs]: Fixed

File: src/ServiceManagement/ExpressRoute/Commands.ExpressRoute/CrossConnections/GetAzureCrossConnection.cs (hunk 3 above)
Comment (nupurkot) [inline src/ServiceManagement/ExpressRoute/Commands.ExpressRoute/CrossConnections/GetAzureCrossConnection.cs]: Fixed throughout

File: setup/azurepowershell.sln (hunk 2 above)
Comment (nupurkot) [inline setup/azurepowershell.sln]: Done.
